src/docs.rs

```diff
@@ -153,7 +171,81 @@ pub fn generate_html(
     files
 }
 
-fn function(statement: &TypedStatement) -> Option<Function<'_>> {
+fn build_source_link_generator(
+    project_root: impl AsRef<Path>,
+    project_config: &PackageConfig,
+    module_path: &PathBuf,
+    name: &String,
+) -> Box<dyn SourceLinkGenerator> {
+    match (&project_config.version, &project_config.repository) {
+        (Some(version), Repository::GitHub { url }) => {
+            let relative_path = module_path
+                .strip_prefix(&project_root)
+                .map(|path| path.to_str())
+                .unwrap_or_default()
+                .unwrap_or_default();
+
+            let src = std::fs::read_to_string(&module_path).unwrap_or_default();
```

Comment:
Rather than read the file again I think we probably have the source code already in memory at a higher level, we can pass it down into this function.

Reply:
I wasn't sure if that was ideal for other compiler operations but if you're happy with that, I can try to look into it.

Reply:
I've done my best here. Passing the src as a string along on the `Analysed` struct. I do end up cloning it though as I'm not sure about the lifetimes where I'm using it. Maybe it can be avoided though.